### ISSUE TITLE
ax.fill_between broken for log scale and values below zero

### ISSUE THREAD

**Issue by hsandmeyer** [matplotlib/matplotlib#9457]
### ax.fill broken for log scale and values below zero

When using ax.fill_between in log scale plots with y-values below zero, the result looks broken and areas are filled that should not be filled

**Code for reproduction**

```python

import numpy as np
import matplotlib.pyplot as plt
xdata=np.arange(0,10,0.01)
ydata = np.exp(-xdata)
edata = 0.2*(10-xdata)*np.cos(5*xdata)*np.exp(-xdata)

fig, ax = plt.subplots()
plt.yscale('log')

ax.fill_between(xdata, ydata - edata, ydata + edata)

plt.savefig("test.png")
```

**Actual outcome**

![broken](https://user-images.githubusercontent.com/26244423/31670601-433a7536-b358-11e7-917c-cd12ce7484aa.png)

**Expected outcome**

![correct](https://user-images.githubusercontent.com/26244423/31670617-4a8b29b6-b358-11e7-82d6-c106a3429f54.png)
The correct example has been generated with matplotlib-2.0.2-1

Probably related: #8623
How to deal with invalid values in log scale plots is certainly a matter of definition. However, it should never happen that areas are filled that should not be filled, as it is the case in the example above.

**Matplotlib version**

  * Operating system: arch linux
  * Matplotlib version: 2.1.0-1
  * Matplotlib backend: Qt5Agg
  * Python version: 3.6.2
  * Other libraries: numpy version: 1.13.3-1

Everything installed using the package manager of arch linux.

**Comment by anntzer** [comment]
Probably can be fixed by a variant of https://github.com/matplotlib/matplotlib/pull/9305...

**Comment by tacaswell** [comment]
Fixed by #9477
